WifiART/src/structs/ArtSystemState.java

```diff
@@ -1,18 +1,24 @@
 package structs;
 
+import java.util.ArrayList;
+
+/**
+ * The system state enclosing the LCI/LCR parameters.
+ */
 public class ArtSystemState {
 
     // Constants
 
     // Parameters
+    private ArrayList<Subelement> subelements;
```

Comment:
Since the user can change which subelements they want to include by checking and unchecking boxes in the Generate panel, the list of subelements would change after construction. However, I am considering instead storing these as key-value pairs where the key is a string identifying the subelement ("LCI", "BSSID", etc.) and the value is the object whose class implements Subelement. This would allow adding and removing subelements on the fly without worrying about indices. I may settle on a different solution though, as this will become more relevant when I flesh out the Model section of the project.